demo/src/framework/demo-header.ts

```diff
@@ -8,24 +8,134 @@
+
+import type { Settings } from "./types";
 
 /**
  * `DemoHeader` is a custom Lit element representing a header component.
  */
 @customElement("demo-header")
 export class DemoHeader extends LitElement {
+  @property({ attribute: false })
+  accessor settings: Settings | undefined = undefined;
+
+  @property({ attribute: false })
+  accessor chatInstance: ChatInstance | null = null;
+
+  private _clickInProgress = false;
+
+  onClick = async () => {
+    if (this.chatInstance) {
+      const state = this.chatInstance?.getState();
+      if (state.viewState.mainWindow) {
+        await this.chatInstance.changeView(ViewType.LAUNCHER);
+      } else {
+        await this.chatInstance.changeView(ViewType.MAIN_WINDOW);
+      }
+    }
+  };
+
+  private _handleButtonClick = async () => {
+    if (this._clickInProgress) {
+      return;
+    }
+
+    this._clickInProgress = true;
+    this.requestUpdate();
+
+    try {
+      await this.onClick();
+    } finally {
+      this._clickInProgress = false;
+      this.requestUpdate();
+    }
+  };
+
+  private _handlePanelToggle = (event: Event) => {
+    const target = event.currentTarget as HTMLElement | null;
+    if (!target) {
+      return;
+    }
+
+    const panelId = target.getAttribute("panel-id");
+    if (!panelId) {
+      return;
+    }
+
+    const panel = this.renderRoot?.querySelector<HTMLElement>(`#${panelId}`);
+    if (!panel) {
+      return;
+    }
+
+    if (panel.hasAttribute("expanded")) {
+      panel.removeAttribute("expanded");
+    } else {
+      panel.setAttribute("expanded", "");
+    }
+  };
+
   render() {
+    const chatInstanceReady = this.chatInstance ? true : false;
+    const isFullscreen = Boolean(
+      this.settings?.layout === "fullscreen" ||
+        this.settings?.layout === "fullscreen-no-gutter",
+    );
+
     return html`
       <cds-header
         aria-label="Carbon AI Chat"
-        style="height: 40px; box-sizing: content-box;"
+        style="${isFullscreen ? `height: 40px; box-sizing: content-box;` : ""}"
       >
         <cds-header-name
           href="https://chat.carbondesignsystem.com/tag/latest/docs/documents/Overview.html"
           prefix="Carbon"
           >AI chat</cds-header-name
         >
+        <div class="cds--header__global">
+          ${chatInstanceReady
+            ? html`<cds-header-global-action
+                aria-label="Open AI Chat"
+                tooltip-text="Open AI Chat"
+                ?disabled=${this._clickInProgress}
+                @click=${this._handleButtonClick}
+              >
+                ${iconLoader(AiLaunch20, { slot: "icon" })}
+              </cds-header-global-action>`
+            : ""}
+          <cds-header-global-action
+            aria-label="Open Resources Panel"
+            tooltip-text="Open Resources Panel"
+            tooltip-alignment="right"
+            panel-id="switcher-panel"
+            @click=${this._handlePanelToggle}
+          >
```

Comment:
```suggestion
          ${chatInstanceReady
            ? html`<cds-header-global-action
                aria-label="Open AI Chat"
                tooltip-text="Open AI Chat"
                ?disabled=${this._clickInProgress}
                @click=${this._handleButtonClick}
                @keydown=${(e: KeyboardEvent) => {
                  if (e.key === 'Enter') {
                    this._handleButtonClick();
                  }
                }}
              >
                ${iconLoader(AiLaunch20, { slot: "icon" })}
              </cds-header-global-action>`
            : ""}
          <cds-header-global-action
            aria-label="Open Resources Panel"
            tooltip-text="Open Resources Panel"
            tooltip-alignment="right"
            panel-id="switcher-panel"
            @click=${this._handlePanelToggle}
            @keydown=${(e: KeyboardEvent) => {
              if (e.key === 'Enter') {
                this._handlePanelToggle(e);
              }
            }}
          >
```

Adding some keyboard support for the new global actions